server 3.4.0 release (#6446)
circleci/circleci-docs: jekyll/_cci2/server-3-whats-new.adoc

toc::[]

== Release 3.4.0

> === Upgrade/install notes

Author: ```suggestion
=== Upgrade notes
```

> * Before installing or upgrading to v3.4.0, delete the deployment named `circleci-server-kube-state-metrics`.

Author: ```suggestion
Before upgrading to v3.4.0, delete the deployment named `circleci-server-kube-state-metrics`:

* Run `kubectl delete deployment/circleci-server-kube-state-metrics --cascade=orphan`
* Redeploy the app from the KOTS admin console
```

toc::[]

== Release 3.4.0

=== Upgrade notes

Before upgrading to v3.4.0, delete the deployment named `circleci-server-kube-state-metrics`:

* Run `kubectl delete deployment/circleci-server-kube-state-metrics --cascade=orphan`

> * Redeploy the app from the KOTS admin console

Reviewer: ```suggestion
* Deploy the app from the KOTS admin console
```

If the user deletes the deployment before hitting Deploy then there's no need to hit Redeploy

Author: Thanks!

=== New Features

* Android images for the machine executor are now available. For more information, see the https://circleci.com/docs/2.0/android-machine-image/[Using Android Images with the Machine Executor] guide.
* The CircleCI Build Agent can now be pulled from a custom Docker registry if required. For more information contact customer support.
* You can now use Amazon Certificate Manager (ACM) to manage your TLS certificates. For more information see the https://circleci.com/docs/2.0/server-3-install/#frontend-settings[Server v3.x Installation Guide].
* Kubernetes v1.16 - v1.23 are now supported.
* Workload Identities for installations on GCP are now supported. Workload Identities can be used as an alternative to static credentials. For more information see the https://circleci.com/docs/2.0/server-3-install-prerequisites/#enable-workload-identities-in-gke-optional[Server v3.x Installation Guide]

=== Fixes

* Fixed a bug that was causing the jobs page and step output access to run very slow in the CircleCI web app.

> * Fixed a bug that was causing the workflows conductor pod to crash and issues with StatsD metrics.

Reviewer 2: ```suggestion
* Fixed a bug that was causing the workflows conductor pod to crash and cause issues with StatsD metrics.
```
